tests/misc-test.cc

```diff
@@ -240,13 +240,56 @@ namespace
 	}
 } // namespace
 
+void check_sealed_scoping()
+{
+	Capability<void> o{switcher_current_thread()};
+	TEST(o.is_valid() && (o.type() == CheriSealTypeSealedTrustedStacks),
+	     "Shared object cap not as expected: {}",
+	     o);
+
+	// Take the address of the o cap, requiring that it go out to memory.
+	Capability<Capability<void>> oP{&o};
+
+	/*
+	 * Load a copy of our sealed o cap through an authority that lacks
+	 * LoadGlobal permission.  The result should be identical to the original
+	 * but without global permission.
+	 */
+	Capability<Capability<void>> oPNoLoadGlobal = oP;
+	oPNoLoadGlobal.permissions() &=
+	  PermissionSet::omnipotent().without(Permission::LoadGlobal);
+	Capability<void> oLocal1 = *oPNoLoadGlobal;
+
+	TEST(
+	  oLocal1.is_valid() && (oLocal1.type() == o.type()) &&
+	    (oLocal1.permissions() == o.permissions().without(Permission::Global)),
+	  "Loading global sealed cap through non-LoadGlobal authority gone wrong: "
+	  "{} {}",
+	  o,
+	  oLocal1);
+
+	/*
+	 * Use CAndPerm to shed Global from our o cap.
+	 * Spell this a little oddly to make sure we get CAndPerm with a mask of
```

Comment:
I think we should add a `drop_perms` method to `Capability` to do this. The architectural permissions were ordered to optimize dropping specific permissions in this way for common permissions to drop (e.g. LM, LG).